test/cuda/test_corner_cases.cc

```diff
@@ -285,6 +285,17 @@ TEST(TestCornerCases, E23) {
       at::Scalar(d[0]).toFloat());
 }
 
+TEST(TestCornerCases, E24) {
+  auto a = I();
+  auto b = I();
+  auto r = I(0);
+  Succeed(
+      "def f(int32 a, int32 b) -> (c) { c(i) = int32(a % b) where i in 0:1 }",
+      {a, b},
+      {r});
+  CHECK_EQ(at::Scalar(r[0]).toInt(), 0);
```

Comment:
I understand that this file is only meant to check compilation, but shouldn't also add a test
somewhere that checks that the correct operation is being performed?

Reply:
this is the correctness check as well. I can modify it a bit so it actually is visible as correctness test. Currently, I did mental calculation on inputs and added the expected output as 0 but that's not the safe way, I'll change this :)